src/shared/version_op.c

```diff
@@ -760,8 +760,8 @@ os_info *get_unix_version()
                       goto free_os_info;
                   } else {
                       char *base;
-                      char *found;
-                      char tag[] = "Oracle Solaris";
+                      char *found = NULL;
+                      char tag[] = "Solaris";
                       if (found = strstr(buff, tag), found) {
```

Comment:
There is no need to use a comma in this if statement, since the variable "found" is the one you are trying to evaluate in the expression. Using comma discards the result of the first operand, and then evaluates the second operand that is the same as the first.

> In the C and C++ programming languages, the comma operator (represented by the token ,) is a binary operator that evaluates its first operand and discards the result, and then evaluates the second operand and returns this value (and type).

Maybe you can directly use:
`if (found = strstr(buff, tag))`

Reply:
In addition to this, I think the right way to represent this is by using double parenthesis since you're using an assignment in an if statement.

Reply:
I agree with you guys, I moved outside. Thanks !